code/game/mecha/mecha.dm

```diff
@@ -501,6 +501,8 @@
 
 /obj/mecha/attack_animal(mob/living/simple_animal/user as mob)
 	log_message("Attack by simple animal. Attacker - [user].",1)
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.do_attack_animation(src)

```

Comment:
Should the piece of code in `simple_animal` that calls `attack_animal` have this code instead, or is there something I am missing?